### PR TITLE
Various Features

### LINKED ISSUE #14: Implement survey form

### PR DESCRIPTION
**Pull request by lukasbischof**
* implements form 
* correct setup
* improvements in factories and specs

closes #14

### CODE REVIEW

File: app/models/answer.rb [hunk 1]
```diff
@@ -4,6 +4,4 @@ class Answer < ApplicationRecord
   belongs_to :answer_possibility
   belongs_to :survey_entry
   belongs_to :question
-
-  validates :answer_possibility, :question, :survey_entry, presence: true
```
Comment (lukasbischof) [inline app/models/answer.rb]: Not needed since it belongs to them

File: spec/services/answer_group_calculation_service_spec.rb [hunk 2]
```diff
@@ -75,50 +65,43 @@
...
+      answer_possibilities.second.map do |possibility|
+        build(:answer, question: questions.second, answer_possibility: possibility)
+      end
     end
-    # END QUESTION GROUP 2
 
-    let(:expected_hash) { { question_group_1 => 2.0, question_group_2 => 1.0 } }
+    let(:expected_hash) { { question_groups.first => 2.0, question_groups.second => 1.0 } }
 
-    before { create :survey_entry, survey: survey, answers: answers_1 + answers_2 }
```
Comment (lukasbischof) [inline spec/services/answer_group_calculation_service_spec.rb]: One survey entry answered the same question multiple times, which is not particularly valid…

File: app/models/question.rb [hunk 3]
```diff
@@ -3,6 +3,7 @@
 class Question < ApplicationRecord
   has_and_belongs_to_many :answer_possibilities
   has_many :answers, dependent: :destroy
+  has_one :survey, through: :question_group, inverse_of: :questions
```
Comment (TheFehr) [inline app/models/question.rb]: has_one? Can't a question be added to multiple surveys?

Comment (lukasbischof) [reply app/models/question.rb]: Currently not as you would need a joining table. But I have in mind that we discussed that we don‘t want that as you would rather provide a survey duplication feature

**Pull request merged**